app/infrastructure/db/repositories/in_memory_user_repository.py

```
@@ -23,3 +26,19 @@ def create(self, user: User) -> User:
+        for user in self._users.values():
+            if user.id == user_id:
+                return user
+        return None
+
+    def update_profile(self, user_id: str, full_name: str | None) -> User | None:
+        user = self.get_by_id(user_id)
+        if not user:
+            return None
+        
+        # Create a copy with the new name (since User is likely frozen/immutable)
+        updated_user = replace(user, full_name=full_name)
```

Ruff (0.14.14)

[error] Undefined name `replace`

(F821)